[DNM] boards: Add support for TFM in nRF5340 PDK and nRF9160 DK

anangl opened the pull request.

ioannisg commented on boards/arm/nrf9160dk_nrf9160/nrf9160dk_nrf9160_common.dts:
```diff
@@ -190,11 +189,11 @@
 		};
 
-		sram0_bsd: image_bsd@20010000 {
+		sram0_bsd: image_bsd@20016000 {
```
@anangl  I am not sure this will work out-of-the-box
IIRC, the sram0_bsd needs to be on this specific location.
I wonder if we could leave this untouched, or create a sample overlay for the partitions (just to have the samples working properly)

anangl committed:
tfm_integration: Use common routine for post-build commands

Provide in the tfm module a cmake function that will return commands
to be executed after a successful build with the CONFIG_BUILD_WITH_TFM
option enabled, and place there operations (signing and merging of
binary images when BL2 is used) that are currently present separately
in particular boards that support building with TF-M.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
[DNM] west.yml: Update tfm module revision

to make it support nRF5340 and nRF9160.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
[DNM] boards: Add support for TF-M in nRF5340 PDK and nRF9160 DK

Provide the required settings so that it is possible to build with
TF-M as the secure image for those boards.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
[DNM] samples/tfm_integration: Add overlays for nRF5340/nRF9160

Add DT overlays that adjust secure and non-secure RAM partitions
to match TF-M requirements for boards nrf5340pdk_nrf5340_cpuappns
and nrf9160dk_nrf9160ns.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>
